### ISSUE TITLE
Trezor model T not detected as a valid device

### ISSUE THREAD

**Issue by mmahut** [shimunn/fido2luks#3]
When trying fido2luks with my Trezor model T, it is not detected.

### Device and software information

```
$ fido2luks --version
fido2luks 0.2.1 (e7049a281a63e3f72f83b720eb44ccb0c65ed4e1)
```

```
$ trezorctl  get-features
Features (155 bytes) {
    backup_type: Bip39 (0),
    capabilities: [
        Bitcoin (1),
        Bitcoin_like (2),
        Binance (3),
        Cardano (4),
        Crypto (5),
        EOS (6),
        Ethereum (7),
        Lisk (8),
        Monero (9),
        NEM (10),
        Ripple (11),
        Stellar (12),
        Tezos (13),
        U2F (14),
        Shamir (15),
        ShamirGroups (16),
    ],
    device_id: '8FD6A92C73B51F1403A27057',
    flags: 0,
    initialized: True,
    label: 'My Trezor',
    language: 'english',
    major_version: 2,
    minor_version: 1,
    model: 'T',
    needs_backup: True,
    no_backup: False,
    passphrase_cached: False,
    passphrase_protection: False,
    patch_version: 8,
    pin_cached: True,
    pin_protection: True,
    recovery_mode: False,
    revision: 8 bytes b'8eb6ce08',
    unfinished_backup: False,
    vendor: 'trezor.io',
}
```

### Debug information

```
$ strace -ff fido2luks credential

(...)

openat(AT_FDCWD, "/sys/class/hidraw/hidraw1/device/report_descriptor", O_RDONLY|O_CLOEXEC) = 4
fstat(4, {st_mode=S_IFREG|0444, st_size=4096, ...}) = 0
read(4, "\6\320\361\t\1\241\1\t \25\0&\377\0u\10\225@\201\2\t!\25\0&\377\0u\10\225@\221"..., 4097) = 34
read(4, "", 4063)                       = 0
close(4)                                = 0
openat(AT_FDCWD, "/dev/hidraw1", O_RDWR|O_CLOEXEC) = 4
getrandom("", 0, GRND_NONBLOCK)         = 0
getrandom("\x31\x44\xa2\x66\x55\x6f\xb7\x0e\xcd\xbc\xf1\x12\x4e\xdd\xf6\xc3\x14\x2b\xc4\x2e\x2b\x1a\x47\x35\x74\x9e\xa9\x7f\x31\xcf\x2f\x20", 32, GRND_NONBLOCK) = 32
write(4, "\0\377\377\377\377\206\0\10@\305I\313?\245n\321\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 65) = 65
read(4, "\377\377\377\377\206\0\21@\305I\313?\245n\321`Q\24s\2\2\0\0\5\0\0\0\0\0\0\0\0"..., 64) = 64
write(4, "\0`Q\24s\220\0\1\4\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 65) = 65
read(4, "`Q\24s\220\0B\0\244\1\202fU2F_V2hFIDO_2_0\2\201khm"..., 64) = 64
read(4, "`Q\24s\0\365bup\365buv\365\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 64) = 64
close(4)                                = 0
getdents(3, /* 0 entries */, 32768)     = 0
close(3)                                = 0
write(2, "no authenticator found, please e"..., 62no authenticator found, please ensure you device is plugged in) = 62
write(2, "\n", 1
)       
sigaltstack({ss_sp=NULL, ss_flags=SS_DISABLE, ss_size=8192}, NULL) = 0
munmap(0x7fbe3a992000, 8192)            = 0
exit_group(4)                           = ?
+++ exited with 4 +++

```

```
$ lsusb | grep -i trezor
Bus 001 Device 061: ID 1209:53c1 Generic SatoshiLabs TREZOR
```

```
$ ls -la /sys/class/hidraw/hidraw*/device|grep -i 1209:53c1
lrwxrwxrwx 1 root root 0 Jan  8 14:38 /sys/class/hidraw/hidraw1/device -> ../../../0003:1209:53C1.00F9
```

```
$ usbhid-dump

001:061:001:DESCRIPTOR         1578490537.827086
 06 D0 F1 09 01 A1 01 09 20 15 00 26 FF 00 75 08
 95 40 81 02 09 21 15 00 26 FF 00 75 08 95 40 91
 02 C0
```

```
$ lsusb -v -s 001:061:001

Bus 001 Device 061: ID 1209:53c1 Generic SatoshiLabs TREZOR
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.10
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1209 Generic
  idProduct          0x53c1 SatoshiLabs TREZOR
  bcdDevice            2.00
  iManufacturer           1 SatoshiLabs
  iProduct                2 TREZOR
  iSerial                 3 8FD6A92C73B51F1403A27057
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0040
    bNumInterfaces          2
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0x80
      (Bus Powered)
    MaxPower              100mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           2
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              5 TREZOR Interface
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x01  EP 1 OUT
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           2
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              5 TREZOR Interface
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      34
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x02  EP 2 OUT
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
Binary Object Store Descriptor:
  bLength                 5
  bDescriptorType        15
  wTotalLength       0x001d
  bNumDeviceCaps          1
  Platform Device Capability:
    bLength                24
    bDescriptorType        16
    bDevCapabilityType      5
    bReserved               0
    PlatformCapabilityUUID    {3408b638-09a9-47a0-8bfd-a0768815b665}
      WebUSB:
        bcdVersion    1.00
        bVendorCode      1
        iLandingPage     1 https://trezor.io/start
can't get debug descriptor: Resource temporarily unavailable
Device Status:     0x0000
  (Bus Powered)
```

**Comment by shimunn** [comment]
Hi

thanks for your detailed report this particular issue has been fixed in ctap_hmac 0.2.1.
However I still wasn't able to get the Model T to produce a hmac secret which is why I would like you to test if:
https://github.com/shimunn/ctap/blob/f982494d5158062b7ebb9f84cd04f28d2be36ce7/examples/hmac.rs
and 
https://github.com/trezor/trezor-firmware/blob/master/tests/fido_tests/libfido2/hmac-secret.sh
run successfully on your machine & Model T.

**Comment by mmahut** [comment]
@shimunn thank you.

I have tried and both are working fine.

```
 ./hmac-secret.sh
32+0 records in
32+0 records out
32 bytes copied, 0.000219627 s, 146 kB/s
64+0 records in
64+0 records out
64 bytes copied, 0.000368098 s, 174 kB/s
00000000: a239 5759 dbf9 36c9 4b2d 9331 93f7 d5ba  .9WY..6.K-.1....
00000010: abda 3f85 7f31 d5d3 2343 91d9 29e6 9458  ..?..1..#C..)..X
00000020: 6f46 ee81 7810 e7ad f4d8 8d4c 2b2a a1e4  oF..x......L+*..
00000030: 0997 c7b4 ee1a 584c 9530 c53a 2429 acd7  ......XL.0.:$)..
```

```
$ cargo run --bin hmac
warning: variable does not need to be mutable
  --> src/bin/hmac.rs:25:9
   |
25 |     let mut credential = match args().skip(1).next().map(|h| FidoHmacCredential {
   |         ----^^^^^^^^^^
   |         |
   |         help: remove this `mut`
   |
   = note: `#[warn(unused_mut)]` on by default

warning: unused `std::result::Result` that must be used
  --> src/bin/hmac.rs:53:5
   |
53 |     stdout().flush();
   |     ^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_must_use)]` on by default
   = note: this `Result` may be an `Err` variant, which should be handled

    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/hmac`
Authorize using your device
Credential: f1d00200c6ce28af84542d3cf93de0f3b599b27a2682aacd788b27d38f3c2389fe41280fb616c5d012ffbeb399c39ef7b008c79a73e2f58a0a06e0efb2cd436b729e9a6d517091107993397690a915491d1696e6
Note: You can pass this credential as first argument in order to reproduce results
Type in your message: test
Authorize using your device
Hash: 175fc169c2f9870a87b0af5e639499911b6bf15e0f2b11e4e76f86c363a75a0f

```

**Comment by mmahut** [comment]
I can confirm this has been fixed with ```ctap_hmac 0.2.1```.

```
$ fido2luks connected
Found 1 devices

$ $ /home/mmahut/.cargo/bin/fido2luks credential
f1d0020015ec52c30deeec3407175d132c8a34f77fa2788ec0cea440cfea2759ed7a0a2b45eae81f811051666cd95fd5231701d4
```

Feel free to close this issue once released. Thank you!
